### ISSUE TITLE
File path error when running through singularity/apptainer

### ISSUE THREAD

**Issue by steob92** [VERITAS-Observatory/Eventdisplay_AnalysisScripts_VTS#92]
When defining temporary file paths, we're not checking if for if the analysis will be ran through apptainer:
https://github.com/VERITAS-Observatory/Eventdisplay_AnalysisScripts_VTS/blob/7694c0a185bc6e46a1aeb193fc0686b1a48467a1/scripts/helper_scripts/ANALYSIS.anasum_sub.sh#L209-L213

As a result apptainer will search `/opt/TEMPDIR/path/to/local/tmp/dir/` for temporary run parameters and runlists. 

Need to check before running anasum with something like:
```
if [ -n "$EVNDISP_APPTAINER" ]; then
    RUNP="/opt/TEMPDIR/$(basename $RUNP)"
    FLIST="/opt/TEMPDIR/$(basename $FLIST)"
fi
```

There might be similar issues for `evndisp` and `mscw` stage processing. 
ToDo:

- [ ] Check helper scripts for similar issues.

**Comment by steob92** [comment]
This seems to resolve the anasum issue: [4cba8aa](https://github.com/VERITAS-Observatory/Eventdisplay_AnalysisScripts_VTS/commit/4cba8aaa4c8ca539ea7ad102be1e810b25afbc98)

mscw stage also seems to run ok, no changes needed. I am getting errors with evndisp relating to:
```VERITAS_DATA_DIR_2=/lustre/fs23/group/veritas/data```

This looks related to DESY preprocessing and it is getting filled from:
https://github.com/VERITAS-Observatory/Eventdisplay_AnalysisScripts_VTS/blob/4cba8aaa4c8ca539ea7ad102be1e810b25afbc98/scripts/ANALYSIS.evndisp.sh#L224

Since I don't want to break preprocessing, I've added a check to see if this directory exists:
https://github.com/VERITAS-Observatory/Eventdisplay_AnalysisScripts_VTS/blob/012c3e8927c2ed93e4bbece230455575e3ab41ee/scripts/helper_scripts/ANALYSIS.evndisp_sub.sh#L38-L40

I have some DB issue at the moment, but at least the pre-DB stage of `evndisp` is running ok. @GernotMaier are the DB dumps available anywhere for direct download? Also let me know if you have a better solution for my `VERITAS_DATA_DIR_2` issue.

**Comment by GernotMaier** [comment]
I am trying to understand the issue with TEMPDIR: this is set as variable both in the script and in the apptainer environment. I really want to avoid if statements about apptainers in the body of the script - we should try and handle this add the top where we bind the directories.  There must be another issue or this is implemented incorrectly. What is your error message?

**Comment by steob92** [comment]
The job submits and starts on the cluster just fine, however this is the output file I get:
```
VERITAS Analysis Summary (University of Delaware & DESY)  (version v.4.90)
==========================================================================

sequentiell analysis

no file found to read run parameters: /opt/VERITAS_EVNDISP_AUX_DIR//ParameterFiles///raid/jacko/obriens/VERITAS/tmpdir/MSCWDISP-5431c997-03e3-41b3-8016-963f60dc35b8/ANASUM.runparameter
```

You can see that the temp runparameter is getting created where I point my $TMP to, but the filename that is passed to the container is `/opt/VERITAS_EVNDISP_AUX_DIR//ParameterFiles/` (which is the bind point) + `/raid/jacko/obriens/VERITAS/tmpdir/MSCWDISP-5431c997-03e3-41b3-8016-963f60dc35b8/ANASUM.runparameter` (which is where ANASUM.runparameter sits on the host.

Instead it should be:
`/opt/TEMPDIR/ANASUM.runparameter`

Alternatively if `$VERITAS_EVNDISP_AUX_DIR` is bound to `/opt/VERITAS_EVNDISP_AUX_DIR`  then only the `basename` needs to be passed rather than the full path. 

Fixing this, I get a similar issue for `FLIST`.

**Comment by GernotMaier** [comment]
Reading through this, it looks to me it is a mixup of env variables defined on the host system with those in the container.

Reading through https://singularity-userdoc.readthedocs.io/en/latest/environment_and_metadata.html , it looks like singulariy has also a `--cleanenv` command line option. Do you want to try this?

**Comment by steob92** [comment]
I tried with `--cleanenv` by aliasing `alias apptainer="singularity --cleanenv"` at the top of the script.

I'm not sure the issue is with the host system. Looking here:
https://github.com/VERITAS-Observatory/Eventdisplay_AnalysisScripts_VTS/blob/73fdf0ec961e841333321efd870caf621f66d4ce/scripts/helper_scripts/ANALYSIS.evndisp_sub.sh#L249-L258

`$TEMPDIR/$RUN.root` is being explicitly passed to `evndisp` however it is getting set to a path on the host, without getting reset to somewhere within the container:
https://github.com/VERITAS-Observatory/Eventdisplay_AnalysisScripts_VTS/blob/73fdf0ec961e841333321efd870caf621f66d4ce/scripts/helper_scripts/ANALYSIS.evndisp_sub.sh#L25-L31

**Comment by GernotMaier** [comment]
Yes - this is by design. `TEMPDIR` is defined both in the script environment and in the container. Important is to have th `""`, so that the variable is not expanded.

**Comment by GernotMaier** [comment]
Closing this - last activity in June 2024. Please reopen if this is still an issue.
